x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/util/Check.java

```diff
@@ -36,4 +36,15 @@ public static void notNull(Object object, String message, Object... values) {
             throw new SqlIllegalArgumentException(message, values);
         }
     }
+
+    public static void isFixedNumberAndInRange(Object object, String objectName, Long from, Long to) {
+        if ((object instanceof Number) == false || object instanceof Float || object instanceof Double) {
+            throw new SqlIllegalArgumentException("A fixed point number is required for [{}]; received [{}]", objectName,
+                object.getClass().getTypeName());
+        }
+        Long longValue = ((Number) object).longValue();
+        if (longValue < from || longValue > to) {
+            throw new SqlIllegalArgumentException("[{}] equals [{}], out of the allowed range [{}..{}]", objectName, longValue, from, to);
```

Comment:
Unless we use this format elsewhere [{} .. {}] (with two dots), I would use the comma format [{}, {}].

Reply:
Also, I think the style of the error message we use is first state the problem, then the actual value. Something like "[start] out of the allowed range [ ], received [{}]". Found one example (maybe there are more with out of range values) in [the Analyzer](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java#L529).

Reply:
Good point, updated it.